DataCollector/GitCollector.php

```diff
@@ -32,7 +44,11 @@ public function __construct($rootDir)
     public function collect(Request $request, Response $response, \Exception $exception = null)
     {
         try {
-            $repository = new Repository($this->rootDir);
+            $gitBinary=null;
+            if($this->gitBinaryPath){
+                $gitBinary = new GitBinary($this->gitBinaryPath);
+            }
+            $repository = new Repository($this->rootDir,$gitBinary);
```

Comment:
As the default path of `GitBinary` is set to null, you can remove your conditionnal.

```php
$gitBinary = new GitBinary($this->gitBinaryPath);
$repository = new Repository($this->rootDir, $gitBinary);
```